BUG: Frequency not set on empty series

sahildua2305 opened the pull request:
- [x] closes #14320 
- [x] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

sahildua2305 committed:
BUG: Set frequency for empty Series; fixes #14320

Add test to verify frequency for empty series

Update test to compare empty series behaviour
against normal series

Add entry in whatsnew 0.20.0

jreback commented on pandas/tests/frame/test_timeseries.py:
```diff
@@ -323,6 +323,10 @@ def test_asfreq_datetimeindex(self):
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        s = Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H')
+        self.assertEqual(s.index.freq, 'H')
```
construct an expected series and test against that

sahildua2305 replied:
Done! Please check.

jreback reviewed (changes requested):
add a release note in 0.20

sahildua2305 committed:
Fix lint issue; exceeded number of chars on a line

Tarrasch committed:
docs: Remove old warning from dsintro.rst (#14365)

The warning is about something that have been fixed for almost 3 years. Every time a new user excited about pandas start reading the docs, they have to waste brain-cycles ignoring that big red warning bubble.

MattRijk committed:
TST: #14345 fixes TestDatetimeIndexOps test_nat AssertionErrors on 32-bit

closes #14345

Author: mattrijk <matthewrijk@gmail.com>

Closes #14347 from MattRijk/issue#14345 and squashes the following commits:

73ad3d7 [mattrijk] line 2710 formatting update for whitespace
ea4f151 [mattrijk] second formatting update for whitespace
262492f [mattrijk] TST: #14345 fixes TestDatetimeIndexOps test_nat AssertionErors

jreback commented on doc/source/whatsnew/v0.20.0.txt:
```diff
@@ -79,3 +79,5 @@ Performance Improvements
 Bug Fixes
 ~~~~~~~~~
+
+- Bug in ``asfreq``, where frequency wasn't set for empty series (:issue:`14320`)
```
series -> ``Series``

jreback commented on pandas/tests/frame/test_timeseries.py:
```diff
@@ -323,6 +323,12 @@ def test_asfreq_datetimeindex(self):
         tm.assertIsInstance(ts.index, DatetimeIndex)
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        empty_series = Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H')
```
add the issue number as a comment

jreback commented on pandas/tests/frame/test_timeseries.py:
```diff
@@ -323,6 +323,12 @@ def test_asfreq_datetimeindex(self):
 
+    def test_asfreq_datetimeindex_empty_series(self):
+        empty_series = Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H')
+        normal_series = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
```
expected =

jreback commented on pandas/tests/frame/test_timeseries.py:
```diff
@@ -323,6 +323,12 @@ def test_asfreq_datetimeindex(self):
+        empty_series = Series(index=pd.DatetimeIndex([]), data=[]).asfreq('H')
+        normal_series = Series(index=pd.DatetimeIndex(["2016-09-29 11:00"]),
+                                                      data=[3]).asfreq('H')
+        self.assertEqual(empty_series.index.freq, normal_series.index.freq)
```
use ``tm.assert_series_equal``

sahildua2305 replied:
@jreback But we actually want to compare the frequency of the index not the complete series. Also, if we have one empty and one non-empty series, how can we use `assert_series_equal` method?

jorisvandenbossche committed:
DOC: fix formatting issue with msgpack table

tom-bird committed:
DOC: to_csv warns regarding quoting behaviour for floats #14195 (#14228)

Added a small warning that if `float_format` is set then floats will be quoted even if csv.QUOTE_NONNUMERIC is set

jorisvandenbossche committed:
DOC: add 0.19.1 whatsnew file (#14366)

Commit:
BUG: Fix concat key name

closes #14252
Fixes a bug where `pd.concat` didn't propagate the names of keys used to
create  a hierarchical index.

Author: Ben Kandel <ben.kandel@gmail.com>

Closes #14292 from bkandel/fix_concat_key_name and squashes the following commits:

cdc76f6 [Ben Kandel] take out _ensure_index
4a301f8 [Ben Kandel] put back in Index coercion
d8e2c17 [Ben Kandel] remove coercion to Index before _ensure_index
44932cc [Ben Kandel] changed whatsnew entry to 0.19.1
c51df19 [Ben Kandel] _ensure_index
b54b081 [Ben Kandel] simplified logic
3256119 [Ben Kandel] typo
789ecd4 [Ben Kandel] use _ensure_index
350e724 [Ben Kandel] simplified logic
9615a69 [Ben Kandel] extra tests
5c0108b [Ben Kandel] comments
dd3c4cc [Ben Kandel] comments
5cd8392 [Ben Kandel] added test for names
bc5f1fb [Ben Kandel] cleanup
ef6db68 [Ben Kandel] BUG: Propagate key names in concat.

jreback committed:
change impl details slightly for #14292

jreback committed:
DOC: add whitespace to v0.19.1 bug fix section

bkandel committed:
BUG: set_levels set illegal levels. (#14236)

`MultiIndex.set_levels`, when passed in illegal level values, raises an error.
When `inplace=True`, though, the illegal level values are still accepted. This
commit fixes that behavior by checking that the proposed level values are legal
before setting them.

chris-b1 committed:
BLD/CI: cython cache pxd files (#14363)

jreback committed:
DOC: formatting in basics.rst

jreback committed:
BUG: Bug in localizing an ambiguous timezone when a boolean is passed

closes #14402
closes #14405

adamchainz committed:
Convert readthedocs links for their .org -> .io migration for hosted projects (#14406)

As per [their blog post of the 27th April](https://blog.readthedocs.com/securing-subdomains/) ‘Securing subdomains’:

> Starting today, Read the Docs will start hosting projects from subdomains on the domain readthedocs.io, instead of on readthedocs.org. This change addresses some security concerns around site cookies while hosting user generated data on the same domain as our dashboard.

Test Plan: Manually visited all the links I’ve modified.

sahildua2305 committed:
Move whatsnew entry from 0.20.0 to 0.19.1

sahildua2305 committed:
Nit: Updated test mentioning GitHub issue

jreback commented:
@sahildua2305 you need to rebase on master

``git rebase -i origin/master``

sahildua2305 commented:
Oops! Last time I rebased this branch it was fine. I will do that again.

jreback commented:
if that doesn't work, then close and open a new PR. We had shifted the url of the repo.

sahildua2305 commented:
Sure!
